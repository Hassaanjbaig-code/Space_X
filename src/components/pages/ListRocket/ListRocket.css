.rocket {
  padding: 24px;
  width: 100%;
  text-align: start;
}

.rocket img {
  width: 250px;
  height: 150px;
}

.rocket-description {
  padding-left: 12px;
}

.rocket button {
  width: fit-content;
  background-color: blue;
  padding: 8px;
  color: white;
  border: none;
  border-radius: 4px;
}
