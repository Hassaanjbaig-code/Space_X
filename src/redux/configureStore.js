import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketReducer';
import missionReducer from './missions/missionReducer';

export default configureStore({
  reducer: {
    rocketReducer,
    mission: missionReducer.reducer,
  },
});
