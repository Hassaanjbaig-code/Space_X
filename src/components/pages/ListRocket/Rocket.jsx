import React from 'react'
import { useDispatch } from 'react-redux'
import { reserveRocket, unreserveRocket } from '../../../redux/rockets/rocketReducer'
import './ListRocket.css'
import PropTypes from 'prop-types';

const Rocket = (props) => {
  const dispatch = useDispatch()
  const {id, flickr_images, rocket_name, 
    description, reserved} = props.rocket

  
  return (
    <div className="row rocket">
      <img src={flickr_images} alt={`Rocket ${rocket_name}`} />
      <div className="column rocket-description">
        <h4>{rocket_name}</h4>
        <p>{reserved && (<span className='badge-reserved'>Reserved</span>)} {description}</p>
        {reserved? 
         <button onClick={()=>{dispatch(unreserveRocket(id))}}>Cancel reservation</button>
        : <button onClick={()=> {dispatch(reserveRocket(id))}}>Reserve Rocket</button>}
      </div>
    </div>
  );
};

Rocket.propType = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    flickr_images: PropTypes.string,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
  }),
};
export default Rocket;
