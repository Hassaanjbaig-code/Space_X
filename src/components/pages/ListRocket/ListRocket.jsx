import React, { useEffect } from 'react';
import './ListRocket.css';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import Rocket from './Rocket';
import { fetchRockets } from '../../../redux/rockets/rocketReducer';

const ListRocket = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  const listRocket = useSelector((state) => state.rocketReducer.listRocket, shallowEqual);

  return (
    <ul>
      {(listRocket)? listRocket.map(rocket => (
        <li key={rocket.id} ><Rocket rocket={rocket}/></li>
      )): <p>Empty list</p>}
    </ul>
  );
};

export default ListRocket;
