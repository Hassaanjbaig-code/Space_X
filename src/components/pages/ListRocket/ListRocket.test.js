
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { Provider, useDispatch } from 'react-redux';
import React from 'react';
import ListRocket from './ListRocket';
import configureStore from '../../../redux/configureStore';

const store = configureStore
describe('test render', () => {
  test('List Rocket should match snapshoot', () => { 
    jest.mock('../../../redux/rockets/rocketReducer')
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={store}>
          <ListRocket />
        </Provider>
    </React.StrictMode>,
    ).toJSON()
    expect(tree).toMatchSnapshot();
  })
})

