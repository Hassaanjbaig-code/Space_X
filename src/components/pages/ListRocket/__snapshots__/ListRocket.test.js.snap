// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<ul>
  <p>
    Empty list
  </p>
</ul>
`;
